google-cloud-error_reporting/lib/google/cloud/error_reporting.rb

```diff
@@ -0,0 +1,258 @@
+      #              discovered when running from GCP environments.
+      #   * service_name: An identifier for the running service. Optional,
+      #              automatically discovered when running from Google App
+      #              Engine Flex. Otherwise default to "ruby".
+      #   * service_version: A version identifier for the running service.
+      #                      Optional, automatically discovered when running
+      #                      from Google App Engine Flex.
+      #
+      # Note the project_id and keyfile configuration changes won't be picked up
+      # after the first Google::Cloud::ErrorReporting.report call.
+      #
+      # @example
+      #   # in app.rb
+      #   Google::Cloud::ErrorReporting.configure do |config|
+      #     config.project_id = "my-project-id"
+      #     config.keyfile = "/path/to/keyfile.json"
+      #     config.service_name = "my-service"
+      #     config.service_version = "v8"
+      #   end
+      #
+      #   begin
+      #     fail "boom"
+      #   rescue => exception
+      #     # Report exception using configuration parameters provided above
+      #     Google::Cloud::ErrorReporting.report exception
+      #   end
+      #
+      # @return [Stackdriver::Core::Configuration] The configuration object
+      #   the Google::Cloud::ErrorReporting module uses.
+      #
+      def self.configure
+        yield Google::Cloud.configure.error_reporting if block_given?
+
+        Google::Cloud.configure.error_reporting
+      end
+
+      ##
+      # Provides an easy-to-use interface to Report a Ruby exception object to
+      # Stackdriver ErrorReporting service. This method helps users to
+      # transform the Ruby exception into an Stackdriver ErrorReporting
+      # ErrorEvent gRPC structure, so users don't need to. This should be the
+      # prefered method to use when users wish to report captured exception in
+      # applications.
+      #
+      # This public method creates a default Stackdriver ErrorReporting client
+      # and reuse that between calls. The default client is initialized with
+      # parameters defined in {.configure}.
+      #
+      # The error event can be customized before reporting. See the example
+      # below and {Google::Cloud::ErrorReporting::ErrorEvent} class for avaiable
+      # error event fields.
+      #
+      # @example Basic usage
+      #   # in app.rb
+      #   Google::Cloud::ErrorReporting.configure do |config|
+      #     config.project_id = "my-project-id"
+      #     config.keyfile = "/path/to/keyfile.json"
+      #     config.service_name = "my-service"
+      #     config.service_version = "v8"
+      #   end
+      #
+      #   begin
+      #     fail "boom"
+      #   rescue => exception
+      #     # Report exception using configuration parameters provided above
+      #     Google::Cloud::ErrorReporting.report exception
+      #   end
+      #
+      # @example The error event can be customized if needed
```

Comment:
Should be fixed in https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/1454/commits/f7c436756604ca46d2e752c29162a94ea7b554ab